### PR TITLE
partial study update

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package org.gridsuite.modification.server;
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java:1]: License, author

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java [hunk 2]
```diff
@@ -49,20 +54,26 @@ void changeSwitchState(UUID networkUuid, String switchId, String open) {
         sw.setOpen(Boolean.parseBoolean(open));
 
         networkStoreService.flush(network);
+
+        return listener.getModifications();
     }
 
-    boolean applyGroovyScript(UUID networkUuid, String groovyScript) {
+    public Pair<Boolean, Set<String>> applyGroovyScript(UUID networkUuid, String groovyScript) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:61]: Use a dedicated object for this to have a nicer API

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java:7]: Corrected

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationService.java]: Done

**Review by niconoir — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    private void storeModification(Identifiable identifiable) {
+        if (identifiable instanceof Switch) {
+            substationsIds.add(((Switch) identifiable).getVoltageLevel().getSubstation().getId());
+        } else if (identifiable instanceof Injection) {
+            substationsIds.add(((Injection) identifiable).getTerminal().getVoltageLevel().getSubstation().getId());
+        } else if (identifiable instanceof Branch) {
+            substationsIds.add(((Branch) identifiable).getTerminal1().getVoltageLevel().getSubstation().getId());
+            substationsIds.add(((Branch) identifiable).getTerminal2().getVoltageLevel().getSubstation().getId());
+        } else if (identifiable instanceof ThreeWindingsTransformer) {
+            substationsIds.add(((ThreeWindingsTransformer) identifiable).getTerminal(ThreeWindingsTransformer.Side.ONE).getVoltageLevel().getSubstation().getId());
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java:35]: A three windings transformer usually links voltage levels inside one substation.
I can handle the other sides to be more robust.

[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/DefaultNetworkStoreListener.java:35]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 4]
```diff
@@ -26,26 +30,25 @@
     @Inject
     private NetworkModificationService networkModificationService;
 
-    @PutMapping(value = "/networks/{networkUuid}/switches/{switchId}")
-    @ApiOperation(value = "change a switch state in the network")
+    @PutMapping(value = "/networks/{networkUuid}/switches/{switchId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "change a switch state in the network", response = List.class)
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: Corrected

**Review by niconoir — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/NetworkModificationService.java [hunk 5]
```diff
@@ -39,8 +41,11 @@ private Network getNetwork(UUID networkUuid) {
         }
     }
 
-    void changeSwitchState(UUID networkUuid, String switchId, String open) {
+    public Set<String> changeSwitchState(UUID networkUuid, String switchId, String open) {
```
Comment (jonenst) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:44]: In this case, I think the code is better by not using listeners, instead just return if the switch was modifed or not

Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationService.java:44]: Modification done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**